E2E regression test for Audit Trails CSV download

devvaansh opened the pull request:
## Description

Implements automated E2E regression test for the Audit Trails CSV download functionality that previously required manual  testing during Angular 20 migration. This test ensures the feature remains stable across future framework updates and eliminates manual verification overhead.

The test suite validates the complete workflow: login → navigation to Audit Trails → data table loading → CSV download trigger → absence of error notifications.


Additional housekeeping:

Removed obsolete browser versions from .browserslistrc (IE 11, Android 4.4.3-4.4.4)
Refined .gitignore to exclude only Cypress-generated artifacts (downloads, screenshots, videos)

## Related issues and discussion

# WEB-493


Test Coverage
The new audit-trails.cy.ts test suite includes:

-User authentication flow
-Navigation to System > Audit Trails
-Data table population verification
-CSV download action trigger
-Error notification absence validation
-This prevents regression of the CSV download failure that occurred during Angular 20 migration and provides confidence for future Angular 21+ upgrades.


## Screenshots, if any
<img width="429" height="661" alt="Screenshot 2025-12-12 at 1 03 32 AM" src="https://github.com/user-attachments/assets/1113f10f-b718-4ad6-9098-c080c8c8a9e8" />


## Automated video~

https://github.com/user-attachments/assets/c1952f73-41e1-4b84-b239-86fd8c6cde05


## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [X] If you have multiple commits please combine them into one commit by squashing them.

- [X] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Tests**
  * Added an end-to-end test for the Audit Trails flow: login, dismiss post-login warning, global search, navigate to Audit Trails, export CSV, and verify no error notifications.

* **Chores**
  * Removed multiple legacy browser exclusion rules from compatibility settings.
  * Enabled experimental test-studio support in the test runner.
  * Refined test artifact handling to ignore specific test output subdirectories (downloads, screenshots, videos).

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

IOhacker reviewed (approved):
LGTM

gkbishnoi07 commented:
Please squash your commit and run npx prettier --write for these files

Checking formatting...
[warn] cypress.config.ts
[warn] cypress/e2e/spec.cy.ts
[warn] e2e/cypress/tsconfig.json
[warn] Code style issues found in 3 files. Run Prettier with --write to fix.

devvaansh committed:
test: add Cypress E2E test for Audit Trails CSV download

- Added comprehensive E2E test to prevent CSV download regression
- Implemented stable error toast verification to avoid false positives
- Updated .browserslistrc to remove outdated browser versions
- Refined .gitignore to exclude only Cypress-generated artifacts
- Configured proper test selectors and wait strategies

Addresses: WEB-493

devvaansh commented:
@gkbishnoi07 done

gkbishnoi07 commented:
@devvaansh thanks for the PR
